### PR TITLE
Python 3.4/5/6 and PyPy support

### PR DESCRIPTION
**Pull request by jamadden**
- 100% test coverage
- Enable coveralls
- Rename .txt -> .rst
- Test configure.zcml; add a 'browser' extra to enable loading browser.zcml

### CODE REVIEW

File: src/zope/app/broken/browser.zcml [hunk 1]
```diff
@@ -1,14 +1,34 @@
...
+<configure
+    xmlns="http://namespaces.zope.org/zope"
+    xmlns:zcml="http://namespaces.zope.org/zcml"
+    xmlns:browser="http://namespaces.zope.org/browser"
+    i18n_domain="zope">
+
+  <include package="zope.browserresource" file="meta.zcml" />
+  <include package="zope.browserpage" file="meta.zcml" />
+
+  <configure zcml:condition="not-installed zope.app.menus.zmi_views">
```
Comment (jamadden) [inline src/zope/app/broken/browser.zcml]: This doesn't work the way I thought it did. I need to back it out.

**Comment by mgedmin** [comment]
You have PyPI rights.  I've no energy to review the diff today.

File: src/zope/app/broken/broken.py [hunk 2]
```diff
@@ -24,33 +23,32 @@
...
+    def __get_state(self, key, default=None):
+        get = getattr(self.__Broken_state__, 'get', None)
+        if get is None:
+            get = lambda k, d: default
+        return get(key, default)
 
-    __parent__ = property(__parent__)
+    @property
+    def __parent__(self):
+        return self.__get_state('__parent__')
```
Comment (mgedmin) [inline src/zope/app/broken/broken.py:41]: It seems that this now handles more brokenness than before.  This may be worth mentioning in the changelog.

**Review by mgedmin — APPROVED** [review]
LGTM

File: src/zope/app/broken/broken.py (hunk 2 above)
Comment (jamadden) [reply src/zope/app/broken/broken.py:41]: Good point.

**Comment by jamadden** [comment]
Thank you!

**Pull request merged**